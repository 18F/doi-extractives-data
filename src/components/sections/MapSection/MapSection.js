import React, { useEffect, useRef }  from 'react'
import ReactDOM from 'react-dom'
import { useStaticQuery, graphql } from "gatsby"

import * as d3 from 'd3'

import LocationSelector from '../../selectors/LocationSelector'
import Select from '../../selectors/Select'
import styles from './MapSection.module.scss'
 




const MapSection = ({props}) => {

/*
    const data=useStaticQuery(graphql`
          query RevenueTrendsQuery {
        allMonthlyRevenuesByFiscalYear: allResourceRevenuesMonthly(
          filter: {RevenueCategory: {ne: null}}, 
          sort: {fields: [RevenueDate], order: DESC}) {
          group(field: FiscalYear) {
            fiscalYear: fieldValue
            data: edges {
              node {
                id
                FiscalYear
                Revenue
                RevenueCategory
                RevenueDate
                RevenueType
              }
            }
          }
        }
      }
`) 


    console.debug(data);
*/
    
    console.debug(props)
    let states=props.data.states_data.states
    console.debug(states);
    let options=states.map((value)=>{ return {value:value.state.frontmatter.unique_id,
					      name:value.state.frontmatter.title}})
    console.debug(options);
    
    let offshore_regions =props.data.offshore_data.offshore_regions
return (
        <section className={styles.root}>
	<div className={styles.container +' container-page-wrapper'}>
	<div className={styles.containerLeft}>
	<h3>Learn about extractive industries in each state</h3>
	<p>Explore production, revenue, and disbursements data for each state.</p>
	<LocationSelector
    default='Choose location'
    states={states}
    offshore_regions={offshore_regions}
	/>
	<Select options={states} multiple={false} />
    </div>
	<div className={styles.containerRight}>right </div>

    </div>
    </section>
)
}
export default MapSection
