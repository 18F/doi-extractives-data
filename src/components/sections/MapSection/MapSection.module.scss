
.root {
	margin:80px 0px;
}

.title {

}

.revenueTable {
}

.alignRight {

}

.container {
    display: flex;
    justify-content: space-between;
    flex-wrap: nowrap;
}

.containerLeft {
    max-width: 27%;
}

.containerRight {
     width: 70%;
}
