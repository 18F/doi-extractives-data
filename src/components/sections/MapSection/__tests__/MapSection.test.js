import React from 'react'
import renderer from 'react-test-renderer'
import MapSection from '../MapSection'


describe('MapSection', () => {
    it('renders correctly', () =>
       {
	   const tree = renderer
		 .create(<MapSection />)
		 .toJSON()
	   expect(tree).toMatchSnapshot()
  })
})
