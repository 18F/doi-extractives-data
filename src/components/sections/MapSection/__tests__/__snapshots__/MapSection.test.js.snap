// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapSection renders correctly 1`] = `
<section
  className="root"
>
  <div
    className="container container-page-wrapper"
  >
    <div
      className="containerLeft"
    >
      <h3 />
      <p />
      <select
        className="select-dark-gray"
        defaultValue="Choose location"
        id="location-selector"
        onChange={[Function]}
      >
        <option
          disabled={true}
          value="Choose location"
        >
          Choose location
        </option>
        <optgroup
          label="States"
        />
        <optgroup
          label="Offshore Regions"
        />
      </select>
    </div>
    <div
      className="containerRight"
    >
      <div
        className="map"
      />
       
    </div>
  </div>
</section>
`;
